streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -575,8 +584,10 @@ public void close() throws ProcessorStateException {
     void recycle() {
         log.debug("Recycling state for {} task {}.", taskType, taskId);
 
-        final List<TopicPartition> allChangelogs = getAllChangelogTopicPartitions();
-        changelogReader.unregister(allChangelogs);
+        if (!stateUpdaterEnabled) {
+            final List<TopicPartition> allChangelogs = getAllChangelogTopicPartitions();
+            changelogReader.unregister(allChangelogs);
+        }
```

Comment:
Note that once we have only the state updater recycling a state manager becomes a noop.

Reply:
+100